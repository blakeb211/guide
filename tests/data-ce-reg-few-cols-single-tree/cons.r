## Function for predicting least-squares mean of INTRDVX
## Code produced by GUIDE 41.2 on 9/9/23 at 17:4
predicted <- function(){
 if(!is.na(REFGEN) & REFGEN <= 4.50000000000 ){
   if(!is.na(INC_RANK) & INC_RANK <= 0.840186250000 ){
     catvalues <- c("1","4","6")
     if(INCNONW1 %in% catvalues){
       nodeid <- 8
       predict <- 4147.85464809
     } else {
       nodeid <- 9
       predict <- 1416.34186107
     }
   } else {
     catvalues <- c("1")
     if(INCNONW1 %in% catvalues){
       nodeid <- 10
       predict <- 41147.2485814
     } else {
       nodeid <- 11
       predict <- 9023.60503037
     }
   }
 } else {
   if(!is.na(INC_RANK) & INC_RANK <= 0.936266900000 ){
     if(!is.na(INC_RANK) & INC_RANK <= 0.818551050000 ){
       nodeid <- 12
       predict <- 394.089385435
     } else {
       nodeid <- 13
       predict <- 899.969487084
     }
   } else {
     if(!is.na(AGE_REF) & AGE_REF <= 34.5000000000 ){
       nodeid <- 14
       predict <- 1091.91271527
     } else {
       nodeid <- 15
       predict <- 6061.89855279
     }
   }
 }
 return(c(nodeid,predict))
}
## end of function
##
##
## If desired, replace "ce2021.txt" with name of file containing new data
## New file must have at least the same variables with same names
## (but not necessarily the same order) as in the training data file
## Missing value code is converted to NA if not already NA
newdata <- read.table("ce2021.txt",header=TRUE,colClasses="character")
## node contains terminal node ID of each case
## pred contains predicted value of each case
node <- NULL
pred <- NULL
for(i in 1:nrow(newdata)){
    AGE_REF <- as.numeric(newdata$AGE_REF[i])
    INC_RANK <- as.numeric(newdata$INC_RANK[i])
    INCNONW1 <- as.character(newdata$INCNONW1[i])
    REFGEN <- as.numeric(newdata$REFGEN[i])
    tmp <- predicted()
    node <- c(node,as.numeric(tmp[1]))
    pred <- c(pred,tmp[2])
}
