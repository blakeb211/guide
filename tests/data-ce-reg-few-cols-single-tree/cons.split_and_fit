AGE_REF                                                     
INC_RANK                                                    
INCNONW1                                                    
REFGEN                                                      
