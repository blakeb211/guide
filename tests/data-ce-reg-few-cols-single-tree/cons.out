             GGG   U   U  I  DDDD   EEEE
            G   G  U   U  I  D   D  E
            G      U   U  I  D   D  E
            G  GG  U   U  I  D   D  EEE
            G   G  U   U  I  D   D  E
            G   G  U   U  I  D   D  E
             GGG    UUU   I  DDDD   EEEE
   
 GUIDE Classification and Regression Trees and Forests
 Version 41.2 (Build date: April 9, 2023)          
 Compiled with gfortran 11.3.0 on Ubuntu 22.04.2                       
 Copyright (c) 1997-2023 Wei-Yin Loh. All rights reserved.
 This software is based upon work partially supported by the U.S. Army Research Office,
 National Science Foundation, National Institutes of Health,
 Bureau of Labor Statistics, USDA Economic Research Service, and Eli Lilly.
  
 This job was started on 09/09/23 at 17:04
  
 Least squares regression tree
 No pruning
 Data description file: ce2021reg.dsc
 Training sample file: ce2021.txt
 Missing value code: NA
 Records in data file start on line 2
 3 N variables changed to S
 D variable is INTRDVX
 Piecewise constant model
 Number of records in data file: 3965
 Length of longest entry in data file: 11
 Missing values found in D variable
 Missing values found among categorical variables
 Separate categories will be created for missing categorical variables
 Smallest and largest positive weights are  1.0725E+03 and 9.3902E+04
  
 Summary information for training sample of size 2487 (excluding observations with
 non-positive weight or missing values in d, e, t, r or z variables)
 d=dependent, b=split and fit cat variable using indicator variables,
 c=split-only categorical, i=fit-only categorical (via indicators),
 s=split-only numerical, n=split and fit numerical, f=fit-only numerical,
 m=missing-value flag variable, p=periodic variable, w=weight
                                                  #Codes/           
                                                  Levels/           
  Column  Name            Minimum      Maximum    Periods   #Missing
       3  AGE_REF    s    19.00        87.00    
      31  FINLWT21   w    1072.       0.9390E+05
      48  INC_RANK   s   0.4980E-04    1.000    
      50  INCNONW1   c                                  6     1584
     406  INTRDVX    d    1.000       0.1413E+06
     527  BUSCREEN   c                                  2
     530  REFGEN     s    1.000        5.000    
  
      Total  #cases w/   #missing
     #cases    miss. D  ord. vals   #X-var   #N-var   #F-var   #S-var
       3965       1478          0      543        0        0        3
     #P-var   #M-var   #B-var   #C-var   #I-var
          0        0        0        2        0
 Weight variable FINLWT21 in column: 31
 Number of cases used for training: 2487
 Number of split variables: 5
 Number of cases excluded due to 0 W or missing D variable: 1478
  
 Constant fitted to cases with missing values in regressor variables
 No nodewise interaction tests
 Split values for N and S variables based on exhaustive search
 Maximum number of split levels: 3
 Minimum node sample size: 24
 Top-ranked variables and 1-df chi-squared values at root node
      1  0.8297E+02   REFGEN
      2  0.8111E+02   AGE_REF
      3  0.7066E+02   INCNONW1
      4  0.3558E+02   INC_RANK
      5  0.2731E+01   BUSCREEN

  
 Structure of final tree. Each terminal node is marked with a T.
  
 D-mean is weighted mean of INTRDVX in the node
 Cases fit give the number of cases used to fit node
 MSE is residual sum of squares divided by number of cases in node
        Node    Total    Cases Matrix    Node      Node          Split
        label   cases      fit  rank    D-mean     MSE           variable
           1     2487     2487     1  5.131E+03  8.287E+12   REFGEN 
           2     1914     1914     1  6.476E+03  1.027E+13   INC_RANK 
           4     1345     1345     1  2.823E+03  1.657E+12   INCNONW1 
           8T     702      702     1  4.148E+03  2.667E+12   INC_RANK 
           9T     643      643     1  1.416E+03  4.681E+11   AGE_REF 
           5      569      569     1  1.498E+04  2.828E+13   INCNONW1 
          10T     110      110     1  4.115E+04  6.163E+13   REFGEN 
          11T     459      459     1  9.024E+03  1.594E+13   INC_RANK 
           3      573      573     1  9.804E+02  1.105E+12   INC_RANK 
           6      499      499     1  5.414E+02  8.332E+10   INC_RANK 
          12T     348      348     1  3.941E+02  5.422E+10   BUSCREEN 
          13T     151      151     1  9.000E+02  1.468E+11   AGE_REF 
           7       74       74     1  4.340E+03  7.804E+12   AGE_REF 
          14T      25       25     1  1.092E+03  5.313E+10   - 
          15T      49       49     1  6.062E+03  1.165E+13   INC_RANK 
  
 Number of terminal nodes of final tree: 8
 Total number of nodes of final tree: 15
 Second best split variable (based on curvature test) at root node is AGE_REF
  
 Regression tree:
 For categorical variable splits, values not in training data go to the right
  
 Node 1: REFGEN <= 4.5000000
   Node 2: INC_RANK <= 0.84018625
     Node 4: INCNONW1 = "1", "4", "6"
       Node 8: INTRDVX-mean = 4147.8546
     Node 4: INCNONW1 /= "1", "4", "6"
       Node 9: INTRDVX-mean = 1416.3419
   Node 2: INC_RANK > 0.84018625 or NA
     Node 5: INCNONW1 = "1"
       Node 10: INTRDVX-mean = 41147.249
     Node 5: INCNONW1 /= "1"
       Node 11: INTRDVX-mean = 9023.6050
 Node 1: REFGEN > 4.5000000 or NA
   Node 3: INC_RANK <= 0.93626690
     Node 6: INC_RANK <= 0.81855105
       Node 12: INTRDVX-mean = 394.08939
     Node 6: INC_RANK > 0.81855105 or NA
       Node 13: INTRDVX-mean = 899.96949
   Node 3: INC_RANK > 0.93626690 or NA
     Node 7: AGE_REF <= 34.500000
       Node 14: INTRDVX-mean = 1091.9127
     Node 7: AGE_REF > 34.500000 or NA
       Node 15: INTRDVX-mean = 6061.8986
 
 ***************************************************************
 Predictor means are the weighted means.
  
 WARNING: p-values below not adjusted for split search. For a bootstrap solution see:
  
 1. Loh et al. (2016), "Identification of subgroups with differential treatment effects
 for longitudinal and multiresponse variables", Statistics in Medicine, v.35, 4837-4855.
  
 2. Loh et al. (2019), "Subgroups from regression trees with adjustment for prognostic
 effects and post-selection inference", Statistics in Medicine, v.38, 545-557.
  
 3. Loh and Zhou (2020), "The GUIDE approach to subgroup identification",
 in "Design and Analysis of Subgroups with Biopharmaceutical Applications", Springer, pp.147-165.
  
Node 1: Intermediate node
 A case goes into Node 2 if REFGEN <= 4.5000000
 REFGEN mean = 3.4302980
 Coefficients of least squares regression function:
 Regressor    Coefficient  t-stat      p-value
 Constant      5131.        12.43       0.2220E-15
 INTRDVX mean = 5130.60
 ----------------------------
Node 2: Intermediate node
 A case goes into Node 4 if INC_RANK <= 0.84018625
 INC_RANK mean = 0.64479638
 ----------------------------
Node 4: Intermediate node
 A case goes into Node 8 if INCNONW1 = "1", "4", "6"
 INCNONW1 mode = "1"
 ----------------------------
Node 8: Terminal node
 Coefficients of least squares regression functions:
 Regressor    Coefficient  t-stat      p-value
 Constant      4148.        9.326        0.000    
 INTRDVX mean = 4147.85
 ----------------------------
Node 9: Terminal node
 Coefficients of least squares regression functions:
 Regressor    Coefficient  t-stat      p-value
 Constant      1416.        7.248       0.1221E-11
 INTRDVX mean = 1416.34
 ----------------------------
Node 5: Intermediate node
 A case goes into Node 10 if INCNONW1 = "1"
 INCNONW1 mode = "NA"
 ----------------------------
Node 10: Terminal node
 Coefficients of least squares regression functions:
 Regressor    Coefficient  t-stat      p-value
 Constant     0.4115E+05    7.520        0.000    
 INTRDVX mean = 41147.2
 ----------------------------
Node 11: Terminal node
 Coefficients of least squares regression functions:
 Regressor    Coefficient  t-stat      p-value
 Constant      9024.        6.765       0.4080E-10
 INTRDVX mean = 9023.61
 ----------------------------
Node 3: Intermediate node
 A case goes into Node 6 if INC_RANK <= 0.93626690
 INC_RANK mean = 0.69180398
 ----------------------------
Node 6: Intermediate node
 A case goes into Node 12 if INC_RANK <= 0.81855105
 INC_RANK mean = 0.65545690
 ----------------------------
Node 12: Terminal node
 Coefficients of least squares regression functions:
 Regressor    Coefficient  t-stat      p-value
 Constant      394.1        4.649       0.4747E-05
 INTRDVX mean = 394.089
 ----------------------------
Node 13: Terminal node
 Coefficients of least squares regression functions:
 Regressor    Coefficient  t-stat      p-value
 Constant      900.0        4.156       0.5418E-04
 INTRDVX mean = 899.969
 ----------------------------
Node 7: Intermediate node
 A case goes into Node 14 if AGE_REF <= 34.500000
 AGE_REF mean = 34.198415
 ----------------------------
Node 14: Terminal node
 Coefficients of least squares regression functions:
 Regressor    Coefficient  t-stat      p-value
 Constant      1092.        3.208       0.3765E-02
 INTRDVX mean = 1091.91
 ----------------------------
Node 15: Terminal node
 Coefficients of least squares regression functions:
 Regressor    Coefficient  t-stat      p-value
 Constant      6062.        1.748       0.8684E-01
 INTRDVX mean = 6061.90
 ----------------------------
 Proportion of variance (R-squared) explained by tree model: 0.1851
 
 Observed and fitted values are stored in cons.node
 R code is stored in cons.r
 Split and fit variable names are stored in cons.split_and_fit
 Elapsed time in seconds: 4.9897
